### ISSUE TITLE
If client requests latest page version with an LSN, check that no newer version exists.

### ISSUE THREAD

**Issue by hlinnaka** [neondatabase/neon#626]
Each GetPage request contains 'latest' parameter, and an LSN. When 'latest' is true, the LSN is a hint to the page server that there are no page versions newer than that LSN. We should sanity-check that there indeed aren't any newer page versions, and throw a warning or error if there are.

**Comment by LizardWizzard** [comment]
Related PR is merged, can this be closed? @hlinnaka

**Comment by hlinnaka** [comment]
> Related PR is merged, can this be closed? @hlinnaka

No, this issue is about adding an additional sanity check. When client requests the latest page, and the request includes an LSN, it means that there should not be any newer page version of the page after that LSN. We could check that in the page server, to notice earlier that the client is confused with the requests it's making.

**Comment by stepashka** [comment]
this is blocked till #1338 is resolved and #936 is merged & integrated

**Comment by shanyp** [comment]
@LizardWizzard is this still relevant ?

**Comment by LizardWizzard** [comment]
The context is still valid. We dont have such a check. Not sure if it can be cheaply added. @hlinnaka wdyt?
